src/test/groovy/com/rabbitmq/http/client/ReactiveClientSpec.groovy

```diff
@@ -0,0 +1,1557 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rabbitmq.http.client
+
+import com.rabbitmq.client.AuthenticationFailureException
+import com.rabbitmq.client.Channel
+import com.rabbitmq.client.Connection
+import com.rabbitmq.client.ConnectionFactory
+import com.rabbitmq.http.client.domain.AlivenessTestResult;
+import com.rabbitmq.http.client.domain.BindingInfo;
+import com.rabbitmq.http.client.domain.ChannelInfo;
+import com.rabbitmq.http.client.domain.ClusterId;
+import com.rabbitmq.http.client.domain.ConnectionInfo;
+import com.rabbitmq.http.client.domain.CurrentUserDetails;
+import com.rabbitmq.http.client.domain.Definitions;
+import com.rabbitmq.http.client.domain.ExchangeInfo;
+import com.rabbitmq.http.client.domain.NodeInfo;
+import com.rabbitmq.http.client.domain.OverviewResponse;
+import com.rabbitmq.http.client.domain.PolicyInfo;
+import com.rabbitmq.http.client.domain.QueueInfo
+import com.rabbitmq.http.client.domain.ShovelDetails;
+import com.rabbitmq.http.client.domain.ShovelInfo;
+import com.rabbitmq.http.client.domain.ShovelStatus;
+import com.rabbitmq.http.client.domain.UserInfo;
+import com.rabbitmq.http.client.domain.UserPermissions;
+import com.rabbitmq.http.client.domain.VhostInfo;
+import org.springframework.http.HttpStatus
+import org.springframework.web.reactive.function.client.ClientResponse
+import org.springframework.web.reactive.function.client.WebClientException
+import reactor.core.publisher.Flux
+import spock.lang.IgnoreIf
+import spock.lang.Specification
+
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+class ReactiveClientSpec extends Specification {
+
+    protected static final String DEFAULT_USERNAME = "guest"
+
+    protected static final String DEFAULT_PASSWORD = "guest"
+
+    protected ReactiveClient client
+
+    private final ConnectionFactory cf = initializeConnectionFactory()
+
+    protected static ConnectionFactory initializeConnectionFactory() {
+        final cf = new ConnectionFactory()
+        cf.setAutomaticRecoveryEnabled(false)
+        cf
+    }
+
+    def setup() {
+        client = newLocalhostNodeClient()
+        client.getConnections().toStream().forEach({ c -> client.closeConnection(c.name).block() })
+    }
+
+    protected static ReactiveClient newLocalhostNodeClient() {
+        new ReactiveClient(
+                String.format("http://%s:%s@127.0.0.1:15672/api", DEFAULT_USERNAME, DEFAULT_PASSWORD)
+        )
+    }
+
+    def "GET /api/overview"() {
+        when: "client requests GET /api/overview"
+        final conn = openConnection()
+        final ch = conn.createChannel()
+        1000.times { ch.basicPublish("", "", null, null) }
+
+        def res = client.getOverview().block()
```

Comment:
I know about the `StepVerifier`, but I don't know if it would fit well with Spock tests structure. I kept the reactive client tests as close as possible to the blocking client ones for simplicity sake (there are over 70 tests). Now, I may have a look where the `StepVerifier` could simplify the test. Thanks for the hint!